common/lib/xmodule/xmodule/video_module/transcripts_utils.py

```diff
@@ -729,29 +729,30 @@ def available_translations(self, transcripts, verify_assets=None, include_val_tr
 
         sub, other_langs = transcripts["sub"], transcripts["transcripts"]
 
+        # include val transcripts
+        translations = get_available_transcript_languages(edx_video_id=self.edx_video_id)
+
         # If we're not verifying the assets, we just trust our field values
         if not verify_assets:
+
             if other_langs:
                 translations = list(other_langs)
+
             if not translations or sub:
                 translations += ['en']
-            return translations
 
-        # If we've gotten this far, we're going to verify that the transcripts
-        # being referenced are actually either in the contentstore or in edx-val.
-        if include_val_transcripts:
-            translations = get_available_transcript_languages(edx_video_id=self.edx_video_id)
+            return list(set(translations))
 
         if sub:  # check if sjson exists for 'en'.
             try:
-                Transcript.asset(self.location, sub, 'en')
+                get_transcript_for_video(
```

Comment:
@muhammad-ammar we can call this in a single loop (by combining sub + other_langs)